torch/_inductor/codegen/simd.py

```diff
@@ -1622,77 +1744,249 @@ def codegen_combo_kernel(self, combo_kernel_node):
-                    # this is not a real split
-                    continue
+    @classmethod
+    def create_tiling(
+        cls, pw_tiling: Sequence[sympy.Expr], reduction_tiling: Sequence[sympy.Expr]
+    ) -> Dict[str, Tuple[sympy.Expr]]:
+        """
+        Create a tiling dict from pointwise and reduction splits.
+        """
+        pw_prefixes = ["z", "y", "x"][-len(pw_tiling) :]
+        reduction_prefixes = ["r0_", "r1_"][: len(reduction_tiling)]
+        return immutable_dict(
+            list(zip(pw_prefixes, pw_tiling))
+            + list(zip(reduction_prefixes, reduction_tiling))
+        )
+
+    @classmethod
+    def create_partial_tiling(
+        cls,
+        tiling: Sequence[sympy.Expr],
+        is_pointwise: bool,
+    ) -> Dict[str, sympy.Expr]:
+        return cls.create_tiling(
+            tiling if is_pointwise else [],
+            tiling if not is_pointwise else [],
+        )
 
-            except ValueError:
+    @classmethod
+    def complete_partial_tiling(
+        cls, tiling: Dict[str, sympy.Expr], node
+    ) -> Dict[str, sympy.Expr]:
+        """
+        Given a tiling for only pointwise or reduction dimensions, adds the missing one.
+        """
+        splits = list(tiling.values())
+        is_pointwise = "x" in tiling
+        missing_ranges = node.get_ranges()[1 if is_pointwise else 0]
+        missing_numel = [sympy_product(missing_ranges)]
+
+        tiling_args = (
+            (splits, missing_numel) if is_pointwise else (missing_numel, splits)
+        )
+        return cls.create_tiling(*tiling_args)
+
+    @classmethod
+    def get_nd_tilings(
+        cls,
+        node_schedule: List[IRNode],
+        is_pointwise: bool,
+    ) -> List[Dict[str, Tuple[sympy.Expr]]]:
+        """
+        Creates N-dimensional tiling candidiates, attempting to simplify loads/stores
+        by tiling the kernel into higher dimensions.
+
+        Returns a list of tilings ranked by dimensionality.
+        """
+        tilings = []
+        for node in EnableReduction.filter(node_schedule):
+            if not isinstance(node, scheduler.SchedulerNode):
                 continue
-            tiled_groups = (
-                V.graph.sizevars.simplify(sympy_product(ranges[:split])),
-                V.graph.sizevars.simplify(sympy_product(ranges[split:])),
-            )
-            # score by number of elements
-            score = V.graph.sizevars.size_hint(
-                sympy_product(
-                    size for size, stride in zip(ranges, strides) if stride != 0
+
+            # If we have modular indexing expressions, try to subdivide ranges into
+            # their parameters.
+            range_idx = 0 if is_pointwise else 1
+            node_ranges = node.get_ranges()
+            node_tilings = [node_ranges[range_idx]]
+            for dep in node.read_writes.reads_and_writes():
+                # Check if the variable ranges coincide with the node ranges.
```

Comment:
This part seems to work for the test cases I tried, but it also feels a bit restrictive. The variables had names like `c0` and `d0`, which didn't coincide with the typical triton range tree prefixes. So here I'm just checking that the variable ranges exactly coincide with node ranges, which is a proxy for `dep.ranges` mapping to mapping to `node.ranges`. 

On second thought, I'm not sure if this is strictly necessary. Maybe we can just attempt to tile with `dep.ranges`, and have the tiling compatibility checks tell us whether it's valid?